http/racing_server_test.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+const { run } = Deno;
+
+import { test } from "../testing/mod.ts";
+import { assert, assertEquals } from "../testing/asserts.ts";
+import { BufReader } from "../io/bufio.ts";
+import { TextProtoReader } from "../textproto/mod.ts";
+
+let server;
+async function startServer(): Promise<void> {
+  server = run({
+    args: ["deno", "--A", "http/racing_server.ts"],
```

Comment:
You don't need to do this as a separate process, you can just launch the server in-process here

Reply:
Totally forgot about `Deno.dial` 😂 Updating.

Reply:
About using the same process, we do not have the ability to close the HTTP server so the test would not terminate...

Reply:
why can't we terminate an http server? does Deno.close(server.rid) not work?

Reply:
Since `server` returned by `serve()` is an async iterator of `ServerRequest`s and does not have the rid field.
(I could still add it to each request or to the async iterator itself but feels like a hack...)

Reply:
Having ServerRequest point to its own server seems reasonable.

Reply:
Brief experiment reveals 2 problems at this moment with this:

1. We cannot simply expose the listener and allow users to directly call `close` on it. Since we do not automatically close connections etc. when the last JS reference is dropped, we have to do some cleanup work.
2. There are too many `panic`s on the Rust backend that essentially would always trigger panic here and crash the process if we call `listener.close()` while there is a pending `accept`.

On the other hand, this feels like a separate issue than the one attempted to be addressed here. Probably we could make it another PR? (after fixes for Rust panics land in deno)